src/metabase/models/setting.clj

```diff
@@ -441,10 +442,11 @@
 (defn register-setting!
   "Register a new Setting with a map of `SettingDefinition` attributes. Returns the map it was passed. This is used
   internally be `defsetting`; you shouldn't need to use it yourself."
-  [{setting-name :name, setting-type :type, default :default, :as setting}]
+  [{setting-name :name, setting-type :type, default :default, :as setting} namespace-sym]
```

Comment:
Why not just make namespace symb another map argument? It seems a little weird to have a big options map as arg 1 and then a plain symbol arg as the second arg

Reply:
I left it as a separate arg as its not really part of the setting but more an annotation but i can add it back in.